tests/garage/sampler/test_ray_sampler.py

```diff
@@ -0,0 +1,82 @@
+from unittest.mock import Mock
+
+import numpy as np
+import ray
+
+from garage.envs.grid_world_env import GridWorldEnv
+from garage.sampler import RaySampler, SamplerWorker
+from garage.tf.envs import TfEnv
+from garage.tf.samplers.on_policy_vectorized_sampler \
+    import OnPolicyVectorizedSampler
+from tests.fixtures.policies.scripted_policy import ScriptedPolicy
+
+
+class TestSampler():
+    """
+    Uses mock policy for 4x4 gridworldenv
+    '4x4': [
+        'SFFF',
+        'FHFH',
+        'FFFH',
+        'HFFG'
+    ]
+    0: left
+    1: down
+    2: right
+    3: up
+    -1: no move
+    'S' : starting point
+    'F' or '.': free space
+    'W' or 'x': wall
+    'H' or 'o': hole (terminates episode)
+    'G' : goal
+    [2,2,1,0,3,1,1,1,2,2,1,1,1,2,2,1]
+    """
+
+    def setup_method(self):
+        ray.init(local_mode=True, ignore_reinit_error=True)
```

Comment:
is this required if i want to use RaySampler in a regular launcher?

Reply:
Its not required for a regular launch. The reason I do this is because while running tests, there is a cap on the amount available to the tests, and the ray needs at least 10mb extra of memory to run. When I run it like this, I am running it locally, so that it runs without creating its 10mb object store, and also without creating sampling workers on separate processes.

Reply:
The sampler will run serially this way